lib/SPIRVProducerPass.cpp

```diff
@@ -1549,6 +1566,24 @@ Type *SPIRVProducerPassImpl::CanonicalType(Type *type) {
     // Nothing
   } else if (type->getNumContainedTypes() != 0) {
     switch (type->getTypeID()) {
+    case Type::PointerTyID: {
```

Comment:
Is this needed? There is a test for is pointer type above. Is this also a relic of opaque pointer transitions?

Reply:
This is a relic of opaque pointer, I have removed it